src/main/java/xyz/nucleoid/bedwars/game/active/shop/BwItemShop.java

```diff
@@ -12,24 +12,209 @@
+                    shop.addItem(new ItemStack(BwItems.BRIDGE_EGG), Cost.ofEmeralds(2));
+                    shop.addItem(new ItemStack(BwItems.MOVING_CLOUD), Cost.ofEmeralds(2));
+                }
+            });
+        }
+        return old(player, game);
+    }
+
+    private static void addNavbar(ShopBuilder shop, ServerPlayerEntity player, BwActive game) {
+        //Creates a page navigation bar at the top of the shop
+        shop.add(ShopEntry.ofIcon(Blocks.END_STONE).withName(new LiteralText("Blocks")).withCost(Cost.free()).onBuy(page -> {
+            player.closeHandledScreen();
+            player.openHandledScreen(BwItemShop.create(player, game, 1));
+        }));
+        shop.add(ShopEntry.ofIcon(Items.GOLDEN_SWORD).withName(new LiteralText("Melee Weapons")).withCost(Cost.free()).onBuy(page -> {
+            player.closeHandledScreen();
+            player.openHandledScreen(BwItemShop.create(player, game, 2));
+        }));
+        shop.add(ShopEntry.ofIcon(Items.IRON_CHESTPLATE).withName(new LiteralText("Armor")).withCost(Cost.free()).onBuy(page -> {
+            player.closeHandledScreen();
+            player.openHandledScreen(BwItemShop.create(player, game, 3));
+        }));
+        shop.add(ShopEntry.ofIcon(Items.IRON_PICKAXE).withName(new LiteralText("Tools")).withCost(Cost.free()).onBuy(page -> {
+            player.closeHandledScreen();
+            player.openHandledScreen(BwItemShop.create(player, game, 4));
+        }));
+        shop.add(ShopEntry.ofIcon(Items.BOW).withName(new LiteralText("Archery")).withCost(Cost.free()).onBuy(page -> {
+            player.closeHandledScreen();
+            player.openHandledScreen(BwItemShop.create(player, game, 5));
+        }));
+        shop.add(ShopEntry.ofIcon(Items.TNT).withName(new LiteralText("Utilities and Potions")).withCost(Cost.free()).onBuy(page -> {
+            player.closeHandledScreen();
+            player.openHandledScreen(BwItemShop.create(player, game, 6));
+        }));
+    }
+
+    private static <T extends Upgrade> void addUpgrade(
+            ShopBuilder shop,
+            PlayerUpgrades upgrades, UpgradeType<T> type,
+            Text name
+    ) {
+        int currentLevel = upgrades.getLevel(type);
+        int nextLevel = currentLevel + 1;
+
+        T nextUpgrade = type.forLevel(nextLevel);
+        if (nextUpgrade != null) {
+            shop.add(ShopEntry.ofIcon(nextUpgrade.getIcon())
+                    .withName(name)
+                    .withCost(nextUpgrade.getCost())
+                    .onBuy(p -> {
+                        upgrades.applyLevel(type, nextLevel);
+                    }));
+        } else {
+            T currentUpgrade = type.forLevel(currentLevel);
+            if (currentUpgrade != null) {
+                shop.add(ShopEntry.ofIcon(currentUpgrade.getIcon()).withName(name));
+            }
+        }
+    }
+    private static ShopUi old(ServerPlayerEntity player, BwActive game){
```

Comment:
We'll probably want to remove the old shop ui to avoid duplication, or at least have some data structure holding all the shop entries by categories such that for this page it is able to merge them.